src/main/java/de/hs_mannheim/informatik/ct/persistence/services/DynamicContentService.java

```diff
@@ -61,9 +64,24 @@
         return out.toByteArray();
     }
 
-    public void writeRoomsPrintOutDocx(List<Room> rooms, OutputStream outputStream, Function<Room, UriComponents> uriConverter) throws IOException, XmlException {
-        try(val document = getRoomsPrintOutDox(rooms, uriConverter)) {
-           document.write(outputStream);
+    public void writeRoomsPrintOutDocx(List<Room> rooms, ZipOutputStream outputStream, Function<Room, UriComponents> uriConverter) throws IOException, XmlException {
+        List<XWPFDocument> ListOfDocuments = getRoomsPrintOutDox(rooms, uriConverter, false);
+
+        int counter = 0;
+        outputStream.putNextEntry(new ZipEntry(rooms.get(0).getBuildingName() + "/"));
+        for (XWPFDocument document : ListOfDocuments) {
+
+            try {
+                ByteArrayOutputStream out = new ByteArrayOutputStream();
+                document.write(out);
+                out.close();
```

Comment:
Just a side note, since you don't need this stream. But closing a stream and using it afterwards with `out.toByteArray()` would usually result in an error. `ByteArrayOutputStream` technically don't need to closed, so it worked here. But in general the best way to use `Closable` resources like streams, is by defining it in a try expression, like so:
``` java
try(ByteArrayOutputStream out = new ByteArrayOutputStream()) {
    // do something with the stream
}
```

This will close the stream after the try block, even if an exception is thrown.